### ISSUE TITLE
Pathogenic variants labelled as likely benign by mity-report

### ISSUE THREAD

**Issue by anjalijain22** [KCCG/mity#14]
Variants classified as [known pathogenic variants on MitoMap ](https://www.mitomap.org/foswiki/bin/view/MITOMAP/ConfirmedMutations) are being annotated as "likely benign" in the "MitoTip_interpretation" column of the report generated by Mity. Below is a small part of the annotation report for two known pathogenic variants -
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/anjali/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/anjali/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">

SAMPLE | HGVS | gene/locus | gene/locus   description | MitoTip_score | MitoTip_percentile | MitoTip_interpretation | anticodon | allele_frequency_mitomap | disease_mitomap
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
20 | m.583G>A | MT-TF | Mt_tRNA | 18.6237 | 0.931515957 | likely benign | . | 0 | MELAS / MM & EXIT
20 | m.616T>C | MT-TF | Mt_tRNA | 17.4742 | 0.834607713 | likely benign | . | 0 | Maternally inherited epilepsy

</body>

</html>

I suspect this is happening because in the mitylib>annot>[mitotip_score_fixed_del.csv](https://github.com/KCCG/mity/blob/master/mitylib/annot/mitotip_score_fixed_del.csv) file, all records are labelled as "likely benign" even pathogenic ones. I'd really appreciate if you could look into this and provide a fix.

Thanks,
Anjali

**Comment by DarioS** [comment]
Is the basic problem that the CSV file was last updated in November 2019? It seems ripe for an update.

**Comment by drmjc** [comment]
Hi, thanks for reporting this issue. I don't have anyone actively working on this project, but this seems a simple enough fix. I'd welcome a pull request with this change added if you're able to do this.

**Comment by DarioS** [comment]
Perhaps I could help instead of Anjali. What does fixed in mitotip_score_fixed_del.csv allude to? Was there some special post-processing applied to the CSV file after it was downloaded from MITOMAP website? If so, what is the post-processing?

**Comment by anjalijain22** [comment]
Hi @DarioS, thanks for volunteering to help out. I was working on this issue and have fixed it. But you have raised a good question and I'd be curious to know if there was any post-processing involved before I created a pull request to resolve this issue.

**Comment by drmjc** [comment]
thanks for offering to help, and @anjalijain22 if you've made any changes, I'm very happy to incorporate these annotation files or code fixes back into the repo. you've probably worked out that the RA that developed this has moved on to a busy PhD in London.

Dario,
mitotip scores are fine as they are for SNPs, but not for deletions.

The raw file [from website](https://mitomap.org/downloads/mitotip_scores.txt) has these first few records:

| Position | rCRS | Alt | MitoTIP_Score | Quartile | Count | Percentage | Mitomap_Status |
|----------|------|-----|---------------|----------|-------|------------|----------------|
| 577      | G    | :   | 20.3833       | Q1       | 0     | 0.000      | Absent         |
| 577      | G    | A   | 17.75         | Q1       | 0     | 0.000      | Absent         |
| 577      | G    | C   | 17.75         | Q1       | 0     | 0.000      | Absent         |
| 577      | G    | T   | 17.75         | Q1       | 0     | 0.000      | Absent         |

Whereas this has been reformatted for use in mity as:

| POS | REF | ALT | MitoTip_score | MitoTip_percentile | CHR | MitoTip_interpretation |
|-----|-----|-----|---------------|--------------------|-----|------------------------|
| 576 | AG  | A   | 20.3833       | 0.96808511         | MT  | likely benign          |
| 577 | G   | A   | 17.75         | 0.86419548         | MT  | likely benign          |
| 577 | G   | C   | 17.75         | 0.86419548         | MT  | likely benign          |
| 577 | G   | T   | 17.75         | 0.86419548         | MT  | likely benign          |

Note the deletion record's POS REF and ALT have to change to be able to match VCF syntax.

While I don't have the script that did the conversion, hopefully it's easy to do using mity's `mitotip_score_fixed_del.csv` file with the VCF-style POS-REF-ALT, with the updated data from the mitotip website.

I've requested admin access to write back to the repo, so in the mean here's the R code to set the pathogenicity calls.

```
mtip <- read.csv("mitylib/annot/mitotip_score_fixed_del.csv")
mtip$MitoTip_interpretation <- "likely benign"
mtip$MitoTip_interpretation[orig$MitoTip_score > 8.44] <- "possibly benign"
mtip$MitoTip_interpretation[orig$MitoTip_score > 12.66] <- "possibly pathogenic"
mtip$MitoTip_interpretation[orig$MitoTip_score > 16.25] <- "likely pathogenic"

write.csv(mtip, "mitylib/annot/mitotip_score_fixed_del.csv", row.names=FALSE)
```

**Comment by drmjc** [comment]
actually...

Thanks @anjalijain22 for the reminder about the [Confirmed Mutations table at Mitomap](https://www.mitomap.org/foswiki/bin/view/MITOMAP/ConfirmedMutations). 

However, just a reminder that the pathogenicity calls are from thresholding the [Mitotip scores](https://mitomap.org/downloads/mitotip_scores.txt).

Since the column in the mity result file is called `MitoTip_interpretation`, mity is actually doing the right thing here, and @DarioS, I bet these Mitotip scores likely have not been updated at all since we made that annotation file.

Assuming that the `Mitomap Confirmed Annotations` have experimental support, then I agree these should trump the _in silico_ mitotip scores.

So I think the question here needs to be rescoped to how to incorporate the `Mitomap Confirmed Annotations` into the mity result files & report.

**Comment by anjalijain22** [comment]
Hi @drmjc, thanks for sharing the R code. The code looks perfectly fine and that's how I approached it as well. However, I noticed that it doesn't define the "orig" dataframe used in line 3 which is why all entries are labelled as "likely benign". (I'll push the fix for this)

To your point about MitoTip scores, I randomly checked some variants against the numbers present in mitotip_score_fixed_del.csv and you're right these numbers haven't changed since the last update. 

Most Mitomap confirmed annotations have experimental support which bump them up to likely path/possibly path despite what the mitotip scores suggest and mity accounts for that in the "status_mitomap" column of the the report.
